benchmark/atoi_impl.h

```diff
@@ -0,0 +1,35 @@
+#include "zoo/swar/SWAR.h"
+
+#include <tuple>
+#include <string.h>
+
+uint64_t calculateBase10(
+    zoo::swar::SWAR<8, __uint128_t> convertedToIntegers
+) noexcept;
+
+uint32_t calculateBase10(
+    zoo::swar::SWAR<8, uint64_t> convertedToIntegers
+) noexcept;
+
+namespace zoo {
+
+/// @brief Loads the "block" containing the pointer, by proper alignment
+/// @tparam PtrT Pointer type for loading
+/// @tparam Block as the name indicates
+/// @param pointerInsideBlock the potentially misaligned pointer
+/// @param b where the loaded bytes will be put
+/// @return a pair to indicate the aligned pointer to the base of the block
+/// and the misalignment, in bytes, of the source pointer
```

Comment:
For absolute clarity, specify the range misalignment can take. It probably is not [-maxint, +maxint]